doc/readme.md

```diff
@@ -0,0 +1,335 @@
+
+Sie werden im Laufe der Praktika verschiedene Assets benötigen. Diese liegen per Default im `assets`-Verzeichnis. Sie können das Standardverzeichnis in der `build.gradle` anpassen.
+
+- Standardpfad für Texturen: `assets/`
+- Standardpfad für Charaktere: `assets/character/`
+- Standardpfad für Level-Texturen: `assets/textures/dungeon/`
+
+## Strukturen
+
+In diesem Abschnitt werden Ihnen die wichtigsten Klassen im Dungeon vorgestellt.
+
+![Struktur ECS](ecs/img/ecs.png)
+
+Game ist die Basisklasse von der alles ausgeht. Die Methode `Game#render` ist die Game-Loop. Die Klassen `Entity`, `Component` und `ECS_System` sind die Implementierungen des ECS.
+
+Die LevelAPI generiert, zeichnet und speichert das aktuelle Level. Mehr zum Thema Level erfahren Sie [hier](../level/readme.md).
+
+*Anmerkung: Das UML ist für bessere Lesbarkeit gekürzt.*
+
+*Anmerkung: Die in Gelb hinterlegten Klassen des UML-Diagramms sind für ein Basisverständnis des Dungeons nicht nötig.*
+
+- `entity` Die Java-Implementierung der Entitäten eines ECS
+- `component` Abstrakte Klasse, jedes Component im ECS leitet hiervon ab
+- `ECS-Systems` Abstrakte Klasse, jedes System im ECS leitet hiervon ab
+- LevelAPI: Kümmert sich darum, dass neue Level erzeugt und geladen werden
+- DungeonCamera: Ihr Auge in das Dungeon
+- libGDXSetup: Bereitet die Anwendung vor, für die Verwendung des Dungeons ist Verständnis für die genaue Funktionalität nicht notwendig
+- Game erstellt die Entitäten, Components und Systeme des ECS und beinhaltet die Game-Loop. Game ist Ihr Einstiegspunkt in das Dungeon
+- Game-Loop: Die Game-Loop ist die wichtigste Komponente des Spieles. Sie ist eine Endlosschleife, welche einmal pro Frame aufgerufen wird. Das Spiel läuft in 30 FPS (also 30 frames per seconds), die Game-Loop wird also 30-mal in der Sekunde aufgerufen. Alle Aktionen, die wiederholt ausgeführt werden müssen, wie zum Beispiel das Bewegen und Zeichnen von Figuren, müssen innerhalb der Game-Loop stattfinden. Das Framework ermöglicht es Ihnen, eigene Aktionen in die Game-Loop zu integrieren. Wie genau das geht, erfahren Sie im Laufe dieser Anleitung.
+*Hinweis: Die Game-Loop wird automatisch ausgeführt, Sie müssen sie nicht aktiv aufrufen.*
+
+Zusätzlich existieren noch eine Vielzahl an weiteren Hilfsklassen, mit denen Sie mal mehr oder mal weniger Kontakt haben werden.
+
+### Game
+
+Game implementiert einige wichtige Methoden:
+
+- `setup` wird zu Beginn der Anwendung aufgerufen. In dieser Methode werden die Objekte (wie die Systeme) initialisiert und konfiguriert, welche bereits vor dem Spielstart existieren müssen. In der Vorgabe wird hier bereits das erste Level geladen, die Systeme angelegt und der Herd initialisert.
+- `render` Ruft die Logiken der Systeme auf.
```

Comment:
```suggestion
- `render` Ruft die Logiken der Systeme auf. Dies ist die die Game-Loop.
```

Reply:
Dumme Frage von ganz draußen: _Ist_ `render()` wirklich die Game-Loop oder wird das nur innerhalb der Game-Loop als zentrale Funktion aufgerufen?

Reply:
`Game` erbt von `ScreenAdapter`, eine libGDX-Klasse, wo die Methode `render()` vorgegeben wird, die wir überschreiben. `render()` wird von libGDX in jedem Frame aufgerufen. Vermutlich wird in libGDX eine Loop sein, auf die wir keinen Zugriff haben, weshalb wir `render()` als unseren Game-Loop benutzen.

Reply:
@Lena241 Kannst Du genau diesen Text aus Deinem Kommentar obendrüber ins Dokument schreiben?

> `Game` erbt von `ScreenAdapter`, eine libGDX-Klasse, wo die Methode `render()` vorgegeben wird, die wir überschreiben. `render()` wird von libGDX in jedem Frame aufgerufen. Vermutlich wird in libGDX eine Loop sein, auf die wir keinen Zugriff haben, weshalb wir `render()` als unseren Game-Loop benutzen.